doc/userguide/rules/rule-types.rst

```diff
@@ -0,0 +1,930 @@
+.. role:: example-rule-action
+.. role:: example-rule-header
+.. role:: example-rule-options
+.. role:: example-rule-emphasis
+
+Rule Types and Categorization
+=============================
+
+Once parsed, Suricata rules are categorized for performance and further
+processing (as different rule types will be handled by specific engine modules).
+The signature types are defined in `src/detect.h
+<https://github.com/OISF/suricata/blob/master/src/detect.h>`_:
+
+.. literalinclude:: ../../../src/detect.h
+    :caption: src/detect.h
+    :language: c
+    :start-after: // rule types documentation tag start: SignatureType
+    :end-before: // rule types documentation tag end: SignatureType
+
+In more human readable terms:
+
+.. list-table:: Suricata Rule Types, and their Engine Analysis Term
+    :header-rows: 1
+
+    * - Rule Type
+      - Code Symbol
+      - Engine-Analysis Representation
+    * - Decoder Events Only
+      - ``SIG_TYPE_DEONLY``
+      - ``de_only``
+    * - Packet
+      - ``SIG_TYPE_PKT``
+      - ``pkt``
+    * - IP Only
+      - ``SIG_TYPE_IPONLY``
+      - ``ip_only``
+    * - IP Only (contains negated address(es))
+      - ``SIG_TYPE_LIKE_IPONLY``
+      - ``like_ip_only``
+    * -  Protocol Detection Only
+      - ``SIG_TYPE_PDONLY``
+      - ``pd_only``
+    * - Packet-Stream
+      - ``SIG_TYPE_PKT_STREAM``
+      - ``pkt_stream``
+    * - Stream
+      - ``SIG_TYPE_STREAM``
+      - ``stream``
+    * - Application Layer Protocol
+      - ``SIG_TYPE_APPLAYER``
+      - ``app_layer``
+    * - Application Layer Protocol Transactions
+      - ``SIG_TYPE_APP_TX``
+      - ``app_tx``
+
+The rule type will impact:
+
+  - To what does the signature action apply, in case of a match (`Action Scope`)
+  - When is the rule matched against traffic (`Inspected`)
+  - Against what the rule matches (`Matches`)
+
+This categorization is done taking into consideration the presence or absence of
+certain rule elements, as well as the type of keywords used. The categorization
+currently takes place in `src/detect-engine-build.c:void SignatureSetType()
+<https://github.com/OISF/suricata/blob/master/src/detect-engine-build.c#L1642-L1704>`_.
+
+The ``SignatureSetType()`` overall flow is described below:
+
+.. image:: intro/OverallAlgoHorizontal-20241127.png
+   :width: 600
+   :alt: A flowchart representing the SignatureSetType function.
+
+The following table lists all Suricata signature types, and how they impact the
+aspects aforementioned.
+
+.. list-table:: Suricata Rule Types
+    :widths: 10, 17, 18, 30, 25
```

Comment:
Review width for last 2 columns, especially.

Reply:
the rtd theme isn't properly rendering `list-table` `widths` properly - it's ignoring this, basically.

The best workaround I've found was using [nested paragraphs](https://sphinx-rtd-theme.readthedocs.io/en/latest/demo/lists_tables.html#tables-with-paragraphs), as it seems that the width is being based on the column header, and this seems to be working.